kahuna/public/js/usage-rights/usage-rights-editor.js

```diff
@@ -6,125 +6,112 @@ import './usage-rights-editor.css!';
+        } else {
+            return multiCat;
+        }
+    });
 
-        ctrl.model = getGroupModel(ctrl.usageRights);
-        ctrl.showRestrictions = angular.isDefined(ctrl.model.restrictions);
-    };
-
-    // setting our initial values
-    editsApi.getUsageRightsCategories().then(cats => {
-        const categoriesCopy = angular.copy(cats);
+    // @return Stream.<Category>
+    const categoryChange$ = observe$($scope, () => ctrl.category).filter(cat => !!cat);
 
-        categoriesCopy.forEach(cat => {
-            cat.properties.forEach((property, i) => {
-                let propertyOptions = property.required ? [] : [{key: 'None', value: null}];
+    // @return Stream.<Category>
+    const category$ = categoryFromUsageRights$.merge(categoryChange$).distinctUntilChanged();
 
-                if (property.options) {
-                    property.options.forEach(option => {
-                        propertyOptions.push({key: option, value: option});
-                    });
+    const model$ = usageRights$.map(urs => {
+        const multiModel = reduceObjectsToArrays(urs.map(ur => ur.data));
+        return Object.keys(multiModel).reduce((model, key) => {
+            if (unique(multiModel[key]).length === 1) {
+                model[key] = multiModel[key][0];
+            }
+            return model;
+        }, {});
+    });
 
-                    cat.properties[i].options = propertyOptions;
-                }
-            });
-        });
+    const savingDisabled$ = category$.combineLatest(category$, cat => cat === multiCat);
+    const forceRestrictions$ = model$.combineLatest(category$, (model, cat) => {
+        const defaultRestrictions =
+            cat.properties.find(prop => prop.name === 'defaultRestrictions');
+        const restrictedProp =
+            cat.properties.find(prop => prop.name === 'restrictions');
 
-        ctrl.categories = categoriesCopy;
-        ctrl.originalCats = categoriesCopy;
-        ctrl.update();
+        return defaultRestrictions || (restrictedProp && restrictedProp.required);
     });
 
-    $scope.$watchCollection(() => ctrl.usageRights, onValChange(() => {
-        ctrl.update();
-    }));
-
-    // TODO: What error would we like to show here?
-    // TODO: How do we make this more synchronous? You can only resolve on the
-    // routeProvider, which is actually bound to the UploadCtrl in this instance
-    // SEE: https://github.com/angular/angular.js/issues/2095
-    ctrl.save = () => {
-        // FIXME [1]: We do this as images that didn't have usagerights in the first place will have
-        // the no rights option, this is to make sure we don't override but rather delete.
-        const data = modelToData(ctrl.model);
+    const showRestrictionsChange$ = observe$($scope, () => ctrl.showRestrictions);
+    const showRestrictions$ = showRestrictionsChange$.combineLatest(
+        forceRestrictions$, model$, (forceRestrictions, showRestrictions, model) => {
```

Comment:
>  are the arguments in the resultSelector function in the correct order

No they were't.

I've renamed the streams - it's an complex situation.
* If there are no restrictions and they are not required => false
* If there are restrictions but they are not required => false - but you can turn them off
* If they are required => require them

Does that make sense - probably better :panda_face: 2 :dragon_face: if not.